kong/db/schema/entities/schemas.lua

```diff
@@ -0,0 +1,40 @@
+local typedefs = require "kong.db.schema.typedefs"
+local schema = require "kong.schema"
+
+---@class kong.db.schema.entities.schema : table
+---
+---@field id         string
+---@field name       string
+---@field created_at number
+---@field updated_at number
+---@field tags       string[]
+---@field type       kong.schema.strategy.name
+---@field schema     string
+
+
+return {
+  name = "schemas",
+  primary_key = { "id" },
+  endpoint_key = "name",
+  admin_api_name = "schemas",
+  generate_admin_api = true,
+  workspaceable = false,
+
+  fields = {
+    { id         = typedefs.uuid },
+    { name       = typedefs.utf8_name },
+    { created_at = typedefs.auto_timestamp_s },
+    { updated_at = typedefs.auto_timestamp_s },
+    { tags       = typedefs.tags },
+    { type       = { type = "string", required = true, one_of = schema.strategies }, },
+    { schema     = { type = "string", required = true, }, },
+  },
+
+  entity_checks = {
+    { custom_entity_check = {
+        field_sources = { "type", "schema" },
+        fn = schema.validate_schema,
+      },
+    },
```

Comment:
Note: because the full list of possible schema types is static and known at startup, we could easily rewrite this to use sub-schemas (each `type` is a sub-schema with its own `custom_validator`). Would that be preferable to using `custom_entity_check`?